clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java

```diff
@@ -325,7 +337,17 @@
                                         Type.BOOLEAN,
                                         false,
                                         Importance.LOW,
-                                        ENABLE_IDEMPOTENCE_DOC);
+                                        ENABLE_IDEMPOTENCE_DOC)
+                                .define(TRANSACTION_TIMEOUT_CONFIG,
+                                        Type.INT,
+                                        60000,
+                                        Importance.LOW,
+                                        TRANSACTION_TIMEOUT_DOC)
+                                .define(TRANSACTIONAL_ID_CONFIG,
```

Comment:
Where do we enforce that the transactionalId is not empty? Maybe we could add a `NonEmptyString` validator?